packages/sdk/src/utils/spinner.ts

```diff
@@ -0,0 +1,59 @@
+import type { Options as OraOptions } from 'ora'
+import ora from 'ora'
+
+const defaultOraOptions: OraOptions = {
+  spinner: 'dots',
+  color: 'cyan',
+  indent: 0,
+  stream: process.stdout,
+}
+
+/**
+ * Methods available to a spinner instance that has already started.
+ */
+export interface SpinnerStarted {
+  success(text?: string): void
+  failure(text?: string): void
+}
+
+/**
+ * Closure that starts a spinner if `enableOutput` is true, and returns a `SpinnerStarted` instance.
+ * Note: the spinner is automatically disabled when `process.env.CI` is defined.
+ * @param enableOutput Whether to enable or disable any output. Useful e.g. for "--print" flags in commands.
+ * @param oraOptions Additional options to pass to `ora` for customizing the spinner.
+ * @returns
+ */
+export function createSpinner(enableOutput = true, oraOptions: Partial<OraOptions> = {}) {
+  const actualOptions = { ...defaultOraOptions, ...oraOptions }
+
+  return (text: string): SpinnerStarted => {
+    if (!enableOutput) {
+      return {
+        success: () => {},
+        failure: () => {},
+      }
+    }
+
+    actualOptions.stream?.write('\n')
```

Comment:
Is that still needed?

Reply:
I'd leave it as it is, so that the spinner doesn't appear too close to the upward text. But I'm open to get rid of that line if you feel like it :) (though we'll need to update the snapshots)